### ISSUE TITLE
`dateparser.search.search_dates` does not work as expected

### ISSUE THREAD

**Issue by angel-langdon** [scrapinghub/dateparser#993]
Given the below example
```python
>>> from dateparser.search import search_dates
>>> search_dates("Validity date: 2021-11-26")

```
It does return `None`, however it should return `[('2021-11-26', datetime.datetime(2021, 11, 26, 0, 0))]`

**Comment by Gallaecio** [comment]
Verified with `master` (141199b477cd1ae215207f4d84d0b9e0bae4bba9).

**Comment by dishantsethi** [comment]
Hi @Gallaecio @angel-langdon,
I tried to debug this issue and found that there was a bug while detecting the "best_language" as the text was detected to be in the "vi" language instead of "en"
<img width="725" alt="Screenshot 2021-10-05 at 1 23 50 AM" src="https://user-images.githubusercontent.com/31399916/135916088-71862143-caea-4ca9-bb05-ea546a56808b.png">

I will be trying to fix this same.

**Comment by dishantsethi** [comment]
Forcefully returning detected language as "en" gives the desired result.
<img width="725" alt="Screenshot 2021-10-05 at 1 25 04 AM" src="https://user-images.githubusercontent.com/31399916/135917824-f8bc817a-a546-464b-8517-2194415b6dfb.png">

@Gallaecio What is the logic defined for language detection?

**Comment by Gallaecio** [comment]
If the issue is the language, then this is a non-issue.

@angel-langdon You can either [explicitly pass the language of the input text to `search_dates`](https://dateparser.readthedocs.io/en/latest/?badge=latest#search-for-dates-in-longer-chunks-of-text) if you know it, or [pass it a better language detection function](https://dateparser.readthedocs.io/en/latest/custom_language_detection.html).

**Comment by angel-langdon** [comment]
@Gallaecio  of course, there is an issue, in this case it does not really matter what language is detected, there is clearly a date in the text `2021-11-26`. You can close the issue, but the problem is still there....

**Comment by dishantsethi** [comment]
@Gallaecio @angel-langdon I just figured out that search_dates is giving the desired result till `"Validity date: 2021-11-12"` After this date, None is returned may be because the default date format is YYYY-DD-MM and we are passing the date in format YYYY-MM-DD.

<img width="725" alt="Screenshot 2021-10-05 at 1 44 29 PM" src="https://user-images.githubusercontent.com/31399916/135985842-5de0a28f-a3e4-4aea-be8e-bce9e97c1c67.png">

Correct me if I am wrong with the default date format.
